src/main/scala/fi/oph/koski/omaopintopolkuloki/OmaOpintoPolkuLokiServlet.scala

```diff
@@ -15,16 +16,51 @@ class OmaOpintoPolkuLokiServlet(implicit val application: KoskiApplication) exte
     )
   }
 
-  get("/whoami") {
-    application.opintopolkuHenkilöFacade.findOppijaByOid(session.oid).map(h =>
-      OmaOpintopolkuLokiHenkiloTiedot(h.hetu, h.etunimet, h.kutsumanimi, h.sukunimi)
+  get("/auditlogs/:hetu") {
+    val requestedHetu = params("hetu")
+
+    val huollettava: Option[Huollettava] = session.user.huollettavat.toList.flatMap {
+      case r: HuollettavienHakuOnnistui => r.huollettavat
+        .filter(_.hetu.contains(requestedHetu))
+      case _ => Seq.empty
+    }.headOption
+
+    val oid = huollettava.map(_.oid.getOrElse("")).getOrElse("")
+
+    renderEither(
+      auditLogs.queryLogsFromDynamo(if (huollettava.isDefined) oid else session.oid)
     )
   }
+
+  get("/whoami") {
+    application.opintopolkuHenkilöFacade.findOppijaByOid(session.oid).map(h => {
+      val huollettavat: List[OmaOpintopolkuLokiHenkiloTiedot] = session.user.huollettavat.toList.flatMap {
+        case r: HuollettavienHakuOnnistui => r.huollettavat
+          .map(hh =>
+            OmaOpintopolkuLokiHenkiloTiedot(
+              hetu = hh.hetu,
+              etunimet = hh.etunimet,
+              kutsumanimi = hh.etunimet, // fixme
+              sukunimi = hh.sukunimi,
+              huollettavat = None
+            )
+          )
+        case _ => Seq.empty
+      }
+      OmaOpintopolkuLokiHenkiloTiedot(
+        h.hetu,
+        h.etunimet,
+        h.kutsumanimi,
+        h.sukunimi,
+        huollettavat = if (huollettavat.isEmpty) None else Some(huollettavat)) // eikö tälle ole abstraktiota?
```

Comment:
Ei taida olla meidän scala-versiossa siistiä. https://stackoverflow.com/questions/28244262/convert-a-list-into-an-option-if-it-is-populated ideoiden perusteella halutessaan voisi myös kikkailla esim.: `Option(huollettavat).filter(_.nonEmpty)` tai `huollettavat.headOption.map(_ => huollettavat)` , mutta eksplisiittinen if on kyllä helpompi lukea mielestäni.